src/PathUtilities.cpp

```diff
@@ -0,0 +1,302 @@
+        {
+            if( FileName[CurrPos] == '\\' || FileName[CurrPos] == '/' ) {
+                SlashPos = CurrPos;
+            }
+            ++CurrPos;
+        }
+        if( SlashPos != String::npos ) {
+            return String(FileName,SlashPos + 1);
+        }
+        return String();
+    }
+
+    String GetDirName(const StringView FileName)
+    {
+        size_t SlashPos = FileName.find_last_of("\\/");
+        if( FileName.npos == SlashPos ) {
+            return String();
+        }else{
+            return String( FileName.substr(0,SlashPos + 1) );
+        }
+    }
+
+    String GetBaseName(const char* FileName)
+    {
+        size_t SlashPos = String::npos, CurrPos = 0;
+        while( FileName[CurrPos] != '\0' )
+        {
+            if( FileName[CurrPos] == '\\' || FileName[CurrPos] == '/' ) {
+                SlashPos = CurrPos;
+            }
+            ++CurrPos;
+        }
+        if( SlashPos != String::npos ) {
+            return String(FileName + (SlashPos + 1),(CurrPos - SlashPos) - 1);
+        }
+        return String(FileName,CurrPos);
+    }
+
+    String GetBaseName(const StringView FileName)
+    {
+        size_t SlashPos = FileName.find_last_of("\\/");
+        if( FileName.npos == SlashPos ) {
+            return String( FileName );
+        }else{
+            return String( FileName.substr(SlashPos + 1,FileName.npos) );
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Absolute and Relative Paths
+
+    Boole IsPathAbsolute(const StringView ToCheck)
+    {
+        return IsPathAbsolute_Windows(ToCheck) || IsPathAbsolute_Posix(ToCheck);
+    }
+
+    Boole IsPathAbsolute_Host(const StringView ToCheck)
+    {
+    #ifdef MEZZ_WINDOWS
+        return IsPathAbsolute_Windows(ToCheck);
+    #else
+        return IsPathAbsolute_Posix(ToCheck);
+    #endif
+    }
+
+    Boole IsPathAbsolute_Posix(const StringView ToCheck)
+    {
+        if( !ToCheck.empty() ) {
+            return ( ToCheck[0] == '/' && ToCheck.find_first_of('\\') == String::npos );
+        }
+        return false;
+    }
+
+    Boole IsPathAbsolute_Windows(const StringView ToCheck)
+    {
+        if( ToCheck.size() >= 3 ) {
+            return ( StringTools::IsAlphaLetter(ToCheck[0]) &&
+                     ToCheck[1] == ':' &&
+                     ( ToCheck[2] == '\\' || ToCheck[2] == '/' ) );
+        }
+        return false;
+    }
+
+    Boole IsPathRelative(const StringView ToCheck)
+    {
+        return !IsPathAbsolute(ToCheck);
+    }
+
+    Boole IsPathRelative_Host(const StringView ToCheck)
+    {
+    #ifdef MEZZ_WINDOWS
+        return IsPathRelative_Windows(ToCheck);
+    #else
+        return IsPathRelative_Posix(ToCheck);
+    #endif
+    }
+
+    Boole IsPathRelative_Posix(const StringView ToCheck)
+    {
+        if( !ToCheck.empty() ) {
+            return ( ToCheck[0] != '/' && ToCheck.find_first_of('\\') == String::npos );
```

Comment:
'\\' is actually a valid file in unix filenames. :(

Perhaps we should keep this to enforce this opinion. Posix lets use anything but NULL and '/' in file names, even garbage like '|', '*', or '-n' are valid filenames despite being bullshit.

Reply:
Fixed.